docs/design/ovs-pipeline.md

```diff
@@ -54,525 +54,606 @@
-    ports:
-    - protocol: TCP
-      port: 80
-```
+You can also dump the flows of a table or a group on a given Node as follows:-
+
+```bash
+# Dump flows of a table.
+kubectl exec -n kube-system <ANTREA_AGENT_POD_NAME> -c antrea-ovs -- ovs-ofctl dump-flows <BRIDGE_NAME> table=<TABLE_NAME> -O Openflow15 [--no-stats] [--names]
 
-This Network Policy is applied to all Pods with the `nginx` app label in the
-`default` Namespace. For these Pods, it only allows TCP traffic on port 80 from
-and to Pods which also have the `nginx` app label. Because Antrea will only
-install OVS flows for this Network Policy on Nodes for which some of the Pods
-are the target of the policy, we have scheduled 2 `nginx` Pods on the same
-Node. They received IP addresses 10.10.1.2 and 10.10.1.3 from the Antrea CNI, so
-you will see these addresses show up in the OVS flows.
-
-## Antrea-native Policies Implementation
-
-In addition to the above tables created for K8s NetworkPolicy, Antrea creates
-additional dedicated tables to support the [Antrea-native policies](../antrea-network-policy.md)
-([AntreaPolicyEgressRuleTable] and [AntreaPolicyIngressRuleTable]).
-
-Consider the following Antrea ClusterNetworkPolicy (ACNP) in the Application tier as an
-example for the remainder of this document.
-
-```yaml
-apiVersion: crd.antrea.io/v1beta1
-kind: ClusterNetworkPolicy
-metadata:
-  name: cnp0
-spec:
-  priority: 10
-  tier: application # defaults to application tier if not specified
-  appliedTo:
-    - podSelector:
-        matchLabels:
-          app: server
-  ingress:
-    - action: Drop
-      from:
-        - podSelector:
-            matchLabels:
-              app: notClient
-      ports:
-        - protocol: TCP
-          port: 80
-  egress:
-    - action: Allow
-      to:
-        - podSelector:
-            matchLabels:
-              app: dns
-      ports:
-        - protocol: UDP
-          port: 53
+# Dump a group.
+kubectl exec -n kube-system <ANTREA_AGENT_POD_NAME> -c antrea-ovs -- ovs-ofctl dump-groups <BRIDGE_NAME> table=<TABLE_NAME> <GROUP_ID> -O Openflow15 [--no-stats] [--names]
 ```
 
-This ACNP is applied to all Pods with the `app: server` label in all
-Namespaces. For these Pods, it drops TCP traffic on port 80 from all
-Pods which have the `app: notClient` label. In addition to the ingress rules,
-this policy also allows egress UDP traffic on port 53 to all Pods with the
-label `app: dns`. Similar to K8s NetworkPolicy, Antrea will only install OVS
-flows for this ACNP on Nodes for which some of the Pods are the target of the
-policy. Thus, we have scheduled three Pods (appServer, appDns, appNotClient)
-on the same Node and they have the following IP addresses:
+where `<TABLE_NAME>` is the name of table name in the pipeline and `<GROUP_ID>`
+is the ID of the group.
 
-- appServer: 10.10.1.6
-- appNotClient: 10.10.1.7
-- appDns: 10.10.1.8
+## Registers
 
-## Tables
+We use various 32-bit OVS registers to carry information throughout the pipeline.
```

Comment:
`NXM_NX_XXREG3` is not 32-bit?

Reply:
No, I remember that it is 128-bit, and we only use it to store IPv6 Endpoint.